lib/pure/asyncdispatch.nim

```diff
@@ -232,6 +232,17 @@ when defined(windows) or defined(nimdoc):
     result.timers.newHeapQueue()
     result.callbacks = initDeque[proc ()](64)
 
+  proc close*(dispatcher: PDispatcher) =
+    ## Closes the Dispatcher, freeing the resources associated with it.
+    ##
+    ## Raises ``ValueError`` if the dispatcher contains registered handles,
+    ## because it indicates lack of appropriate cleanup in the application.
+    if closeHandle(dispatcher.ioPort) == 0:
+      raiseOSError(osLastError())
+    if dispatcher.handles.len != 0:
+      raise newException(ValueError,
```

Comment:
I think both options have pros/cons. In either case, it's unlikely you're going to try to meaningfully handle this exception, so I decided it's better to leave it in a more clean state - that is, with the dispatcher handle being closed. Perhaps the exception should actually be changed to AssertionError?